rm/rm-node/src/main/java/org/ow2/proactive/resourcemanager/utils/RMNodeStarter.java

```diff
@@ -399,14 +433,15 @@ public boolean createNodesAndConnect(final String nodeName) {
         return createdNodeNames;
     }
 
-    private List<Node> createNodes(String nodeName) {
-        List<Node> nodes = new ArrayList<>(workers);
+    private Map<String, Node> createNodes(String nodeName) {
+        Map<String, Node> nodes = new HashMap<>(workers);
```

Comment:
`RMNodeStarter` is the class that is used to start a Node process. All data structures used inside are local to the Node process. 

The key used for the map is the URL of the **worker** nodes. It seems to be unique. Below is the output of the URL associated to the Workers when I start `RMNodeStarter` with `-w 3`:

> URL of node 0 pamr://4096/PA-AGENT_NODE_0
> Start of dataspace configuration for node org.objectweb.proactive.core.node.NodeImpl@71834bce
> Dataspace configuration for node org.objectweb.proactive.core.node.NodeImpl@71834bce successful.
> URL of node 1 pamr://4096/PA-AGENT_NODE_1
> Start of dataspace configuration for node org.objectweb.proactive.core.node.NodeImpl@71834f8f
> Dataspace configuration for node org.objectweb.proactive.core.node.NodeImpl@71834f8f successful.
> URL of node 2 pamr://4096/PA-AGENT_NODE_2

The information displayed in the Scheduler portal are not directly related to the map used on the Node process.